api/users.ts

```diff
@@ -9,9 +9,52 @@ export function authentication({ password, username }) {
         username,
       })
       .then((data) => {
-        dispatch(LogIn(data.data));
+        localStorage.setItem("access_token", data.data.access_token);
+        localStorage.setItem("refresh_token", data.data.refresh_token);
+        dispatch(
+          LogIn({
+            access_token: data.data.access_token,
+            refresh_token: data.data.refresh_token,
+          }),
+        );
       })
       .catch((e) => {
+        // alert(e.response.statusText);
+        dispatch(Unauthorized());
+      });
+  };
+}
+
+export function signUp({
+  username,
+  password,
+  repPassword,
+  lastName,
+  firstName,
+  age,
+}) {
+  return async (dispatch) => {
+    await axios
+      .post("http://localhost:4200/user/signup", {
+        username,
+        password,
+        repPassword,
+        lastName,
+        firstName,
+        age,
+      })
+      .then((data) => {
```

Comment:
same, please use `async/await` syntax
it's generally more readable 
same for all other places